core-to-plc/src/Language/Plutus/CoreToPLC.hs

```diff
@@ -451,41 +451,41 @@ isPrimitiveDataCon dc = dc == GHC.intDataCon
 -- These never seem to come up, rather we get the typeclass operations. Not sure if we need them.
 convPrimitiveOp :: (Converting m) => GHC.PrimOp -> m PLCExpr
 convPrimitiveOp = \case
-    GHC.IntAddOp  -> mkIntFun PLC.AddInteger
-    GHC.IntSubOp  -> mkIntFun PLC.SubtractInteger
-    GHC.IntMulOp  -> mkIntFun PLC.MultiplyInteger
+    GHC.IntAddOp  -> pure $ mkIntFun PLC.AddInteger
+    GHC.IntSubOp  -> pure $ mkIntFun PLC.SubtractInteger
+    GHC.IntMulOp  -> pure $ mkIntFun PLC.MultiplyInteger
     -- check this one
-    GHC.IntQuotOp -> mkIntFun PLC.DivideInteger
-    GHC.IntRemOp  -> mkIntFun PLC.RemainderInteger
-    GHC.IntGtOp   -> mkIntRel PLC.GreaterThanInteger
-    GHC.IntGeOp   -> mkIntRel PLC.GreaterThanEqInteger
-    GHC.IntLtOp   -> mkIntRel PLC.LessThanInteger
-    GHC.IntLeOp   -> mkIntRel PLC.LessThanEqInteger
-    GHC.IntEqOp   -> mkIntRel PLC.EqInteger
+    GHC.IntQuotOp -> pure $ mkIntFun PLC.DivideInteger
+    GHC.IntRemOp  -> pure $ mkIntFun PLC.RemainderInteger
+    GHC.IntGtOp   -> pure $ mkIntRel PLC.GreaterThanInteger
+    GHC.IntGeOp   -> pure $ mkIntRel PLC.GreaterThanEqInteger
+    GHC.IntLtOp   -> pure $ mkIntRel PLC.LessThanInteger
+    GHC.IntLeOp   -> pure $ mkIntRel PLC.LessThanEqInteger
+    GHC.IntEqOp   -> pure $ mkIntRel PLC.EqInteger
     po            -> throwSd UnsupportedError $ "Primitive operation:" GHC.<+> GHC.ppr po
 
 -- Typeclasses
 
 convEqMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convEqMethod name
-    | name == GHC.eqName = mkIntRel PLC.EqInteger
+    | name == GHC.eqName = pure $ mkIntRel PLC.EqInteger
     | otherwise = throwSd UnsupportedError $ "Eq method:" GHC.<+> GHC.ppr name
 
 convOrdMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convOrdMethod name
     -- only this one has a name defined in the lib??
```

Comment:
I think I would consistently use `GHC.getOccString` then.

Reply:
I'm still hoping I'll find the other names defined somewhere... I'd rather use the one from GHC if possible.